### PR TITLE
[https://nvbugs/6676511][fix] Reject unsupported speculative outputs

### PR DESCRIPTION
**Pull request by 2ez4bz**
## Dev Engineer Review

- `validate_request` rejects unsupported speculative-decoding outputs at request admission.
- The validation rejects context logits, generation logits, and log-probability outputs.
- The error reports all unsupported requested options in one `ValueError`.
- Non-speculative requests retain these output options.
- The change removes the unused logger import and runtime warnings.
- No configuration or test-list files changed.

## QA Engineer Review

- Modified test file: `tests/unittest/llmapi/test_sampling_params.py`.
- The tests now use production-compatible `TorchLlmArgs` and `_TestLLM`.
- The updated helper invokes `BaseLLM`’s bound generation-config method.
- No test-list entries were added or removed.
- Test-list coverage is not declared in `tests/integration/test_lists/`, `test-db/`, or `qa/`.
- Verdict: needs follow-up.

## Description

* Why?

Speculative decoding does not support context or generation logits and their derived log probabilities. Allowing these requests to reach the executor can crash forked postprocessing workers and leave serving requests permanently hung.

* What?

Reject unsupported logits and log-probability output options at the LLM request boundary whenever speculative decoding is configured. Preserve these options for non-speculative requests and treat top-0 log-probability requests as explicitly requested outputs.
## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/speculative/spec_sampler_base.py [hunk 1]
```diff
@@ -98,9 +96,25 @@ def validate_request(self, request: LlmRequest) -> None:
...
+        requested_outputs = (
+            ("return_context_logits / prompt_logprobs", request.py_return_context_logits),
+            ("return_generation_logits", request.py_return_generation_logits),
+            ("logprobs", request.py_return_log_probs),
+        )
+        unsupported_outputs = [name for name, requested in requested_outputs if requested]
+        if unsupported_outputs:
+            raise ValueError(
+                "The following output options are not supported with "
+                "one-model speculative decoding: "
```
Comment (mikeiovine) [inline tensorrt_llm/_torch/speculative/spec_sampler_base.py]: Remove "one-model". One-model is the only flavor of spec dec supported now and it should just be referred to as "speculative decoding" to make the docs consistent

**Review by mikeiovine — APPROVED** [review]

File: tensorrt_llm/_torch/speculative/spec_sampler_base.py (hunk 1 above)
Comment (2ez4bz) [reply tensorrt_llm/_torch/speculative/spec_sampler_base.py]: Done.
